assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -388,6 +388,68 @@ const baseSelectors = {
 		return modules[ slug ];
 	} ),
 
+	/**
+	 * Gets module dependency names by slug.
+	 *
+	 * Returns a list of modules that depend on this module.
+	 * Returns `undefined` if state is still loading or if said module doesn't exist.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {(Array|undefined)} An array of dependency module names; `undefined` if state is still loading.
+	 */
+	getModuleDependencyNames: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const module = select( STORE_NAME ).getModule( slug );
+
+		// Return `undefined` if module with this slug isn't loaded yet.
+		if ( module === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found; return `[]` to signify the
+		// "not found" state.
+		if ( module === null ) {
+			return [];
+		}
+
+		// Module is found, return the names of the dependencies
+		const modules = select( STORE_NAME ).getModules();
```

Comment:
Returned modules might be `undefined` which means it is not loaded yet. We need to return undefined in this case:
```suggestion
		const modules = select( STORE_NAME ).getModules();
		if ( modules === undefined ) {
			return undefined;
		}
```

Reply:
Modules can't be undefined at this point due to https://github.com/google/site-kit-wp/blob/fdc6a94a3951234c3b836f685d9ec65fcf37f8bf/assets/js/googlesitekit/modules/datastore/modules.js#L378

Reply:
Ah... yes, you are right. Maybe we should add a comment which says that modules are already resolved at this point, shouldn't we?

Reply:
Addressed.

Reply:
Awesome! Thanks!